diff --git a/tensorflow/core/distributed_runtime/tensor_coding.cc b/tensorflow/core/distributed_runtime/tensor_coding.cc
index 78b56b6d363..13f53a0d3f4 100644
--- a/tensorflow/core/distributed_runtime/tensor_coding.cc
+++ b/tensorflow/core/distributed_runtime/tensor_coding.cc
@@ -82,7 +82,11 @@ void TensorResponse::InitPartial(const RecvTensorResponse& response,
 Status TensorResponse::ParseFrom(Source* source) {
   if (!on_host_) {
     protobuf::io::CodedInputStream input(source->contents());
+#if GOOGLE_PROTOBUF_VERSION >= 3006000
+    input.SetTotalBytesLimit(INT_MAX);  // Unlimited
+#else
     input.SetTotalBytesLimit(INT_MAX, INT_MAX);  // Unlimited
+#endif
 
     // Pre-parse into local storage, then delegate to device.
     if (!meta_.ParseFromCodedStream(&input) || !input.ConsumedEntireMessage()) {
@@ -218,7 +222,11 @@ bool TensorResponse::ParseTensorSubmessage(
 
 bool TensorResponse::ParseFast(Source* source) {
   protobuf::io::CodedInputStream input(source->contents());
-  input.SetTotalBytesLimit(INT_MAX, INT_MAX);  // Unlimited
+#if GOOGLE_PROTOBUF_VERSION >= 3006000
+    input.SetTotalBytesLimit(INT_MAX);  // Unlimited
+#else
+    input.SetTotalBytesLimit(INT_MAX, INT_MAX);  // Unlimited
+#endif
   while (true) {
     auto p = input.ReadTagWithCutoff(127);
     int tag = GetTagFieldNumber(p.first);
